### PR TITLE
[MODORDERS-1095] - Split InventoryManager god class into subclasses

### PR DESCRIPTION
**Pull request by imerabishvili**
## Purpose
https://folio-org.atlassian.net/browse/MODORDERS-1095

## Approach
Split class into 3 classes and update related code.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/inventory/InventoryItemManager.java [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+        .compose(item -> {
+          logger.debug("Posting {} physical item(s) for PO Line with '{}' id", quantity, compPOL.getId());
+          return createItemRecords(item, quantity, requestContext);
+        });
+    } else {
+      return Future.succeededFuture(Collections.emptyList());
+    }
+  }
+
+  public Integer extractTotalRecords(JsonObject json) {
```
Comment (damien-git) [inline src/main/java/org/folio/service/inventory/InventoryItemManager.java:472]: This method can be turned private

File: src/main/java/org/folio/service/inventory/InventoryUtils.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+          } else {
+            promise.fail(result.cause());
+          }
+        }));
+
+    return promise.future();
+  }
+
+  public static Future<String> getEntryTypeValue(ConfigurationEntriesCache configurationEntriesCache,
```
Comment (damien-git) [inline src/main/java/org/folio/service/inventory/InventoryUtils.java:106]: This method should be kept private. Is there any other that could be changed to private after refactoring ?

**Review by damien-git — APPROVED** [review]

File: src/test/java/org/folio/service/inventory/InventoryManagerTest.java [hunk 3]
```diff
@@ -587,68 +587,12 @@ void shouldCheckIfTheHoldingExistsWhenHoldingIdSpecifiedAndIfExistThenReturnHold
...
 
     doReturn(succeededFuture(holdingExp)).when(restClient).getAsJsonObject(any(RequestEntry.class), eq(requestContext));
-    String holdingIdAct = inventoryManager.getOrCreateHoldingsRecord(instanceId, location, requestContext).result();
+    String holdingIdAct = inventoryHoldingManager.getOrCreateHoldingsRecord(instanceId, location, requestContext).result();
 
     assertThat(holdingIdAct, equalTo(holdingIdExp));
     verify(restClient, times(1)).getAsJsonObject(any(RequestEntry.class), eq(requestContext));
   }
 
-  @Test
```
Comment (SerhiiNosko) [inline src/test/java/org/folio/service/inventory/InventoryManagerTest.java:596]: can these tests be un-deleted?

File: src/test/java/org/folio/orders/utils/PoLineCommonUtilTest.java [hunk 4]
```diff
@@ -15,40 +15,40 @@
...
-    });
-    //When
-    PoLineCommonUtil.makePoLinesPending(order.getCompositePoLines());
-    //Then
-    order.getCompositePoLines().forEach(line -> {
-      assertEquals(CompositePoLine.PaymentStatus.FULLY_PAID, line.getPaymentStatus());
-      assertEquals(CompositePoLine.ReceiptStatus.FULLY_RECEIVED, line.getReceiptStatus());
-    });
-  }
+//
```
Comment (SerhiiNosko) [inline src/test/java/org/folio/orders/utils/PoLineCommonUtilTest.java:18]: can these tests be un-commented?

Comment (imerabishvili) [reply src/test/java/org/folio/orders/utils/PoLineCommonUtilTest.java:18]: They needed refactoring to mock a static API however IDEA showed that they were not covering additional lines so didn't spent time

File: src/main/java/org/folio/service/inventory/InventoryItemManager.java (hunk 1 above)
Comment (imerabishvili) [reply src/main/java/org/folio/service/inventory/InventoryItemManager.java]: fixed

File: src/main/java/org/folio/service/inventory/InventoryUtils.java (hunk 2 above)
Comment (imerabishvili) [reply src/main/java/org/folio/service/inventory/InventoryUtils.java]: Restricted access for some more methods

File: src/test/java/org/folio/service/inventory/InventoryManagerTest.java (hunk 3 above)
Comment (imerabishvili) [reply src/test/java/org/folio/service/inventory/InventoryManagerTest.java:596]: They needed refactoring to mock a static API however IDEA showed that they were not covering additional lines so didn't spent time

**Pull request merged**